lib/spack/spack/directives.py

```diff
@@ -520,7 +520,10 @@ def _execute_conflicts(pkg):
 
         # Save in a list the conflicts and the associated custom messages
         when_spec_list = pkg.conflicts.setdefault(conflict_spec, [])
-        when_spec_list.append((when_spec, msg))
+        msg_with_name = None
+        if msg is not None:
+            msg_with_name = f"{pkg.name}: {msg}"
+        when_spec_list.append((when_spec, msg_with_name))
```

Comment:
```suggestion
        msg = f"{pkg.name}: {msg}" if msg is not None else msg
        when_spec_list.append((when_spec, msg))
```

@boomanaiden154 thanks for the PR! This looks great! One quick suggestion for simplifying the message transform by simply overwriting the existing message value.

Reply:
Thanks for the suggestion. Using the python ternary is definitely more elegant. Unfortunately, I tried overwriting the `msg` variable, but it's confusing something somewhere even though it is perfectly valid python. I'm getting the following errors when running the unit tests after applying the exact suggestion:
```
github-actions-v0.4 raised RepoError: cannot load package 'python' from the 'builtin' repository: local variable 'msg' referenced before assignment
github-actions-v0.3 raised RepoError: cannot load package 'python' from the 'builtin' repository: local variable 'msg' referenced before assignment
spack-install raised RepoError: cannot load package 'python' from the 'builtin' repository: local variable 'msg' referenced before assignment
```